tests/virt_autotest/virsh_virtual_network.pm

```diff
@@ -0,0 +1,335 @@
+        }
+        #Destroy br123 test environment
+        my $mac_br123 = script_output("virsh domiflist $guest|grep br123|grep -oE \"[[:xdigit:]]{2}(:[[:xdigit:]]{2}){5}\"");
+        assert_script_run("virsh detach-interface $guest bridge --mac $mac_br123");
+        destroy_standalone;
+        restart_libvirtd;
+        #Confirm HOST BRIDGE NETWORK
+        assert_script_run("ssh root\@$gi_host_bridge 'rcnetwork restart'",           60);
+        assert_script_run("ssh root\@$gi_host_bridge 'ping -c2 -W1 openqa.suse.de'", 60);
+        save_screenshot;
+        assert_script_run("virsh detach-interface $guest bridge --current");
+    }
+    #Destroy HOST BRIDGE NETWORK
+    assert_script_run("virsh net-destroy vnet_host_bridge");
+    save_screenshot;
+
+##NAT BASED NETWORK
+##Prepare Test
+    my $vnet_nated = <<'EOF';
+<network>
+  <name>vnet_nated</name>
+  <bridge name="virbr0"/>
+  <forward mode="nat"/>
+  <ip address="192.168.122.1" netmask="255.255.255.0">
+    <dhcp>
+      <range start="192.168.122.2" end="192.168.122.254"/>
+    </dhcp>
+  </ip>
+</network>
+EOF
+    die "There was not the default(NAT BASED NETWORK) virtual network" if (script_run('virsh net-list --all | grep default') != 0);
+
+    #Create NAT BASED NETWORK
+    assert_script_run("virsh net-info default");
+    assert_script_run("virsh net-dumpxml default |tee libvirt_default.xml");
+    assert_script_run("virsh net-undefine default");
+    assert_script_run("echo '$vnet_nated' >> vnet_nated.xml");
+    assert_script_run("virsh net-create vnet_nated.xml");
+    save_screenshot;
+    upload_logs "vnet_nated.xml";
+    assert_script_run("rm -rf vnet_nated.xml");
+
+    foreach my $guest (keys %xen::guests) {
+        record_info "$guest", "NAT BASED NETWORK for $guest";
+        assert_script_run("virsh attach-interface $guest network vnet_nated --live", 60);
+        script_retry "journalctl | grep dnsmasq-dhcp| tail -1| grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"", delay => 60, retry => 3, timeout => 60;
+        my $gi_vnet_nated = script_output("journalctl | grep dnsmasq-dhcp| tail -1| grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"");
+        #Test NAT BASED NETWORK
+        assert_script_run("ssh root\@$gi_vnet_nated 'rcnetwork restart'",           60);
+        assert_script_run("ssh root\@$gi_vnet_nated 'ping -c2 -W1 openqa.suse.de'", 60);
+        save_screenshot;
+        assert_script_run("virsh detach-interface $guest network --current");
+    }
+    #Destroy NATed Network
+    assert_script_run("virsh net-destroy vnet_nated");
+    save_screenshot;
+
+    #Restore default(NATed Network)
+    my $default_path = "/root/libvirt_default.xml";
+    if (script_run("[[ -f $default_path ]]") == 0) { assert_script_run("virsh net-define $default_path", 60); assert_script_run("rm -rf $default_path"); }
+
+##ROUTED NETWORK
+##Prepare Test
+    my $vnet_routed = <<'EOF';
```

Comment:
Wouldn't be better to have it as configuration file in /data ?

Reply:
Just found a useful link for establishing VM on NAT:
https://fabianlee.org/2019/05/26/kvm-creating-a-guest-vm-on-a-nat-network/  if you need.

Reply:
Here, should be refer to https://www.suse.com/documentation/sles-12/singlehtml/book_virt/book_virt.html#libvirt.networks.virtual about NAT firstly.

Reply:
Yes. Just think you may need workaround if anything bad happens. It seems that the author has encountered no obstacle with his attempt.

Reply:
> Wouldn't be better to have it as configuration file in /data ?

Here, just let the libvirt virtual network configuration file and test code in one file, think that this way easy to code maintain. So, do not have the  configuration file in /data now.

Reply:
In my opinion it is also faster to download those config files from `/data` than typing it every time.

Reply:
So far, we need to wait for ROUTE NETWORK setting(included ROUTE NETWORK Configuration File), after submit new PR to have the configuration files under /data.
Now, This PR still use the typing way.

Reply:
I still don't understand why you want to wait with this for a new PR.

Reply:
Now, we just cover 4 types of libvirt virtual network as 1rst step, next, after confirm route network setting, we will update this test as 2nd step, maybe included enhanced the existed test
code (e.g have the configuration file to /data)

Reply:
@guoxuguang Guess you misuderstood pavel. I agree with him.  Please put these config files to openqa data directory. And instead of cat content to a file on SUT, fetch them directly from openqa to SUT. You can ask wayne for details. He did this during DNS implementation.

Reply:
This is a place that need change.

Reply:
Got it. Will update this as your mentioned.

Reply:
Had saved the libvirt virtual network configuration file  under /date dir, please refer to the verification run link as below:
http://10.67.19.82/tests/265